src/Prometheus/RegistryInterface.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+namespace Prometheus;
+
+use Prometheus\Exception\MetricNotFoundException;
+use Prometheus\Exception\MetricsRegistrationException;
+
+interface RegistryInterface
+{
+    /**
+     * @return MetricFamilySamples[]
+     */
+    public function getMetricFamilySamples(): array;
+
+    /**
+     * @param string $namespace e.g. cms
+     * @param string $name e.g. duration_seconds
+     * @param string $help e.g. The duration something took in seconds.
+     * @param array  $labels e.g. ['controller', 'action']
+     *
+     * @return Gauge
+     * @throws MetricsRegistrationException
+     */
+    public function registerGauge($namespace, $name, $help, $labels = []): Gauge;
```

Comment:
Would it make sense to type the parameters (string and array)? 

This would also apply to the other methods as well. What do you think about it?

Reply:
Yes, but i would then suggest we do the entire library in one go, not just the interfaces. WDYT?

Reply:
Makes sense, but I would prefer to change this independently from the interface because the change might be a bigger MR. I will have a look into the typing on Monday.

Reply:
Yes, my suggestion would be we merge these in and I can look into a separate PR with the typing, I have done it to a couple libraries and its not too hard, its a small lib.

Reply:
Would also introduce phpstan along with that to ensure we are not mismatching the types, sound good?

Reply:
phpstan is already there (within the CI) but just on level 1.

Reply:
ok i'll work on getting it higher and available locally as well over the weekend. And I'll type everything.

Reply:
Really thank you!